### PR TITLE
Refactoring discussion post sorting/filtering

### PR DESCRIPTION
**Pull request by maddiegabriel**
Original problem was [US128857](https://rally1.rallydev.com/#/42960320374d/dashboard?detail=%2Fuserstory%2F601065927259). Also fixes [US129047](https://rally1.rallydev.com/#/42960320374d/dashboard?detail=%2Fuserstory%2F601537702489) and [US129054](https://rally1.rallydev.com/#/42960320374d/dashboard?detail=%2Fuserstory%2F601546174909)

This looks kinda messy because I moved a lot of stuff around! The post sorting/filtering logic is now in [consistent-evaluation-evidence-discussion.js](https://github.com/BrightspaceHypermediaComponents/consistent-evaluation/blob/master/components/left-panel/discussions/consistent-evaluation-evidence-discussion.js) and only happens when you explicitly choose to sort/filter the posts (rather than on each render).

### CODE REVIEW

File: components/left-panel/discussions/consistent-evaluation-evidence-discussion.js [hunk 1]
```diff
@@ -86,33 +89,73 @@ export class ConsistentEvaluationEvidenceDiscussion extends SkeletonMixin(RtlMix
...
+	_countPostFilters() {
+		return this._selectedFilters.includes(filterByThreads) + this._selectedFilters.includes(filterByReplies);
+	}
+
+	_countScoreFilters() {
+		return this._selectedFilters.includes(filterByUnscored) + this._selectedFilters.includes(filterByScored);
+	}
+
+	_filterPosts(e) {
+		const newFilters = [...this._selectedFilters];
```
Comment (jeroach) [inline components/left-panel/discussions/consistent-evaluation-evidence-discussion.js:147]: What does this syntax mean? (the `...`)

Just curious :)

Comment (maddiegabriel) [reply components/left-panel/discussions/consistent-evaluation-evidence-discussion.js:147]: I copied that part over BUT I think it's called the 'spread' operator and it just copies all the elements from `this._selectedFilters` into `newFilters`!

**Review by jeroach — APPROVED** [review]
LGTM

vdiff failures seem related to the issue Viktor and Scy are investigating.

Unit test failure was just a timeout, so that can be kicked again.

**Comment by jeroach** [comment]
I was going to make a comment around adding a demo page and vdiff tests, but I still need to add those for the `d2l-table` logic that I recently added.

I just made a story for that: [US129228](https://rally1.rallydev.com/#/42960320374d/custom/236803223728?Iteration=%2Fiteration%2F450584616044&Name=%5BDiscussions%5D%20Add%20demo%20page%20and%20vdiff%20test%20for%20new%20d2l-table%20logic&Project=%2Fproject%2F42960320374&Release=u&cpoid=42960320374&detail=%2Fuserstory%2F601864688567&rankScope=%2Fiteration%2F450584616044&rankTo=BOTTOM&typeDef=7362609683&fdp=true?fdp=true): [Discussions] Add demo page and vdiff test for new d2l-table logic

I can probably work to add coverage for this logic once a) this PR is merged, and b) once I get the base tests taken care of.

**Comment by maddiegabriel** [comment]
> I was going to make a comment around adding a demo page and vdiff tests, but I still need to add those for the `d2l-table` logic that I recently added.
> 
> I just made a story for that: [US129228]: [Discussions] Add demo page and vdiff test for new d2l-table logic
> 
> I can probably work to add coverage for this logic once a) this PR is merged, and b) once I get the base tests taken care of.

Sounds good!! LMK if I can help when the time comes - never done either of those things :)

**Review by JamieTran — APPROVED** [review]
LGTM, it looks like Viktor and Scy decided to skip the problematic tests so pulling their changes should fix the vdiff failures

**Review by rowansilver — APPROVED** [review]

**Pull request merged**